server/server.go

```diff
@@ -248,6 +252,12 @@ func (s *Server) WithStore(store storage.Store) *Server {
 	return s
 }
 
+// WithManager sets the plugins manager used by the server.
+func (s *Server) WithManager(manager *plugins.Manager) *Server {
```

Comment:
I'm wondering whether we should couple the server (and other plugins to the manager like this...) It might be a cleaner if the dependencies are injected one by one.

Did you consider giving the server a callback that would be invoked whenever the compiler changes?

Reply:
+1 for passing deps like this. 
I haven't thought about that. In this case only the watcher is migrated.